### PR TITLE
Release v0.60.0

### PR DESCRIPTION
**Pull request by elfensky**
Production release of everything on `develop` since **v0.59.0** → bundles **v0.59.1, v0.59.2, v0.60.0**.

## v0.60.0 — Archive Analytics quick-wins (features)
- **Combat Telemetry on `/stats`** (#178) — Friendly Fire Index + Accuracy Trend (cross-season % line charts) and a Shots-per-Planet "big number", fed by telemetry sums already in `getCrossSeasonStats`. Hides when no telemetry exists (currently S157–S159).
- **Player Engagement on `/archives`** (#275) — lazy-loaded per-faction scatter of `players_at_start` vs. day-into-war.
- **Closest Calls on `/archives`** (#273) — top-3 narrowest defend losses (region · faction · % held), hidden when a war had no nail-biters.

## v0.59.2 — routing
- `/sign-in` gets its own page title; removed the post-login `/dashboard → /profile` redirect hop (OAuth callbacks point straight at `/profile`).

## v0.59.1 — chart upkeep
- recharts 3.8.1 → 3.9.0; reworked `ProgressExplainer`'s buffer band off the removed `<Area baseLine>` (visual unchanged).

## Verification
- `lint` · `typecheck` · `test:unit` (1498 passed) · `build` all green on `develop`.
- v0.60.0 vizzes DevTools-verified with real data, 0 console errors.

## No DB migrations
This release adds no Prisma migrations.

## Post-merge steps (per CLAUDE.md §Git Workflow rule 3)
1. Tag **`v0.60.0`** on the merge commit on `main` and push it — the production Docker build only triggers on version tags.
2. Merge `main` back into `develop`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by elfensky — COMMENTED** [review]
## /octo:review - Multi-LLM Code Review

| Severity | Count |
|----------|-------|
| Normal | 2 |
| Nit | 7 |
| Pre-existing | 0 |

_Reviewed by Codex + Gemini + Claude + Perplexity fleet_
_See inline comments for details_

File: package-lock.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
     "name": "helldivers.bot",
-    "version": "0.59.0",
+    "version": "0.59.1",
```
Comment (elfensky) [inline package-lock.json]: [NORMAL] **Lock file version (0.59.1) does not match package.json version (0.60.0)**

Two agents flagged this independently. `package.json` and `CHANGELOG.md` move the release to `0.60.0`, but `package-lock.json` still reflects `0.59.1` (both the top-level `version` and `packages[""].version`). CI tooling or deployment pipelines that read the lockfile version for tagging/labeling will report the wrong release number. A fresh `npm install` in a clean environment will overwrite the checked-in lock with the correct version, making the committed file misleading until then.

_Reviewed by /octo:review (multi-LLM fleet)_

File: .github/workflows/metrics.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ jobs:
         steps:
             # https://github.com/lowlighter/metrics/tree/master/source/plugins/pagespeed
             - name: 'metrics: pagespeed'
-              uses: lowlighter/metrics@v3.34
+              uses: lowlighter/metrics@v4
```
Comment (elfensky) [inline .github/workflows/metrics.yml]: [NIT] **GitHub Action pinned to mutable tag `@v4`**

The `lowlighter/metrics` action is pinned to the major-version tag `@v4`. Mutable tags can be reassigned to malicious commits by compromised maintainers or attackers with write access to the action repository. Pinning to a specific git commit SHA is the recommended supply-chain best practice for CI/CD pipelines.

_Reviewed by /octo:review (multi-LLM fleet)_

File: src/features/stats/computeTelemetryStats.mjs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    const rows = (perSeason ?? []).filter(
+        (s) => Number(s.shots) > 0 || Number(s.kills) > 0,
+    );
+
+    // Friendly Fire Index — accidentals as a percentage of total kills.
+    const friendlyFire = rows
+        .filter((s) => Number(s.kills) > 0)
+        .map((s) => ({
+            season: s.season,
+            value: (Number(s.accidentals) / Number(s.kills)) * 100,
```
Comment (elfensky) [inline src/features/stats/computeTelemetryStats.mjs]: [NIT] **Potential NaN propagation on missing properties**

If `s.accidentals` or `s.hits` is undefined or null, `Number()` yields NaN. This propagates to the percentage ratios and into Recharts client components, which may fail to render or crash. Changing to `Number(s.accidentals ?? 0n)` (and equivalent for `hits`) ensures robust calculations.

_Reviewed by /octo:review (multi-LLM fleet)_

File: src/features/archives/PlayerEngagementChart.jsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+ * absent. Events with no positive player count carry no engagement signal and
+ * are dropped — so a season without player data yields an empty series and the
+ * chart (and its section) hides.
+ *
+ * @param {Array<{enemy:number, start_time:number, players_at_start:number, region:number, type:string}>} events - The season's events.
+ * @param {number|null|undefined} warStart - Unix-seconds anchor for day 0.
+ * @returns {Array<{enemy:number, name:string, color:string, points:Array<object>}>}
+ */
+function buildEngagementSeries(events, warStart) {
+    const withPlayers = (events ?? []).filter((e) => (e.players_at_start ?? 0) > 0);
```
Comment (elfensky) [inline src/features/archives/PlayerEngagementChart.jsx]: [NIT] **`buildEngagementSeries` has no unit tests despite non-trivial logic**

Both other new pure-logic helpers in this PR — `selectClosestCalls` and `computeTelemetryStats` — each have a dedicated test file. `buildEngagementSeries` has comparable complexity: day-offset computation from an anchor, fallback to earliest event when `warStart` is absent, per-faction grouping, and zero-player filtering. The `Math.min` spread edge-case noted separately would be caught by a test with a large or missing `warStart` input. A `buildEngagementSeries.test.mjs` would complete the TDD pattern established for this PR.

_Reviewed by /octo:review (multi-LLM fleet)_

File: src/app/stats/page.jsx [hunk 5]
```diff
@@ -2,6 +2,10 @@ import { getCrossSeasonStats } from '@/db/queries/getCrossSeasonStats.mjs';
 import FactionThreatRanking from '@/features/stats/FactionThreatRanking';
 import WarOutcomes from '@/features/stats/WarOutcomes';
 import SeasonRecords from '@/features/stats/SeasonRecords';
+import RatioTrendChart from '@/features/stats/RatioTrendChart';
```
Comment (elfensky) [inline src/app/stats/page.jsx]: [NIT] **`RatioTrendChart` imported statically, breaking the project's `next/dynamic` pattern**

`FactionHealthChart` and `PlayerEngagementChart` both use a `*Loader.jsx` wrapper with `next/dynamic` + `ssr: false` to keep the Recharts bundle out of the initial page JS. `RatioTrendChart` is imported statically in `stats/page.jsx`, so its Recharts code is always bundled even when `telemetry.seasonsWithTelemetry === 0` and the section is hidden. A `RatioTrendChartLoader.jsx` would restore consistency.

_Reviewed by /octo:review (multi-LLM fleet)_

File: src/features/archives/PlayerEngagementChart.jsx [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    if (withPlayers.length === 0) return [];
+
+    const anchor = warStart ?? Math.min(...withPlayers.map((e) => e.start_time));
+
```
Comment (elfensky) [inline src/features/archives/PlayerEngagementChart.jsx]: [NIT] **`Math.min(...array.map(...))` can throw `RangeError` for large event arrays**

`Math.min(...withPlayers.map((e) => e.start_time))` spreads the full array as function arguments. JS engines impose a per-call argument-count limit (~65K–125K); exceeding it throws `RangeError: Maximum call stack size exceeded`. Current season event counts are small but the defensive fix is trivial: `withPlayers.reduce((m, e) => Math.min(m, e.start_time), Infinity)`.

_Reviewed by /octo:review (multi-LLM fleet)_

File: src/features/stats/computeTelemetryStats.mjs [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+ * keeps polling.
+ *
+ * BigInt sums are narrowed to `Number` here (magnitudes are well under 2^53) so
+ * the result is plain and safe to pass across the server→client boundary to the
+ * chart components — BigInt can't be serialized as a client-component prop.
+ *
+ * @param {Array<{season:number, kills:bigint, accidentals:bigint, shots:bigint, hits:bigint, completed_planets:number}>} perSeason - Per-season aggregates from getCrossSeasonStats.
+ * @returns {{ friendlyFire: Array<{season:number, value:number}>, accuracy: Array<{season:number, value:number}>, shotsPerPlanet: number|null, seasonsWithTelemetry: number }}
+ */
+export function computeTelemetryStats(perSeason) {
```
Comment (elfensky) [inline src/features/stats/computeTelemetryStats.mjs:17]: [NIT] **`seasonsWithTelemetry` count can overstate coverage relative to visible chart data**

`rows` is filtered to seasons where `shots > 0 OR kills > 0`; `seasonsWithTelemetry` is `rows.length`. But `friendlyFire` additionally filters to `kills > 0`, and `accuracy` to `shots > 0`. A season with `shots=0, kills>0` increments the count and appears in `friendlyFire` but is absent from `accuracy`. The UI copy "from the X seasons the bot has polled" could overstate coverage by one or more seasons.

_Reviewed by /octo:review (multi-LLM fleet)_

File: .github/workflows/metrics.yml (hunk 2 above)
Comment (elfensky) [inline .github/workflows/metrics.yml]: [NIT] **`lowlighter/metrics` major version jump (v3.34 → v4) has no migration notes**

v4 is a significant rewrite — input names, plugin configuration keys, and committer behaviour changed between majors. The existing `committer_branch: metrics` and `token: NOT_NEEDED` inputs may be silently ignored or renamed in v4. The metrics job could start failing or pushing to the wrong branch without loud CI signal, since misconfigured inputs are not always fatal. Worth validating the v4 migration guide before merging.

_Reviewed by /octo:review (multi-LLM fleet)_

File: package-lock.json (hunk 1 above)
Comment (elfensky) [reply package-lock.json]: Fixed in 126ffc6 — ran `npm install`; the lockfile now reports `0.60.0` (top-level and `packages[""].version`). Folded into the 0.60.0 release.

File: .github/workflows/metrics.yml (hunk 2 above)
Comment (elfensky) [reply .github/workflows/metrics.yml]: Fixed in 126ffc6 — reverted to `lowlighter/metrics@v3.34`. Note `@v4` is an unreleased dev **branch**, not a release tag (latest release is v3.34); a CI group-bump had silently undone the earlier deliberate v3.34 pin (8ea570c7).

File: src/features/stats/computeTelemetryStats.mjs (hunk 3 above)
Comment (elfensky) [reply src/features/stats/computeTelemetryStats.mjs]: Fixed in 126ffc6 — `Number(s.accidentals ?? 0n)` and `Number(s.hits ?? 0n)` now guard the missing-field case.

File: src/features/archives/PlayerEngagementChart.jsx (hunk 4 above)
Comment (elfensky) [reply src/features/archives/PlayerEngagementChart.jsx]: Fixed in 126ffc6 — extracted into `src/features/archives/buildEngagementSeries.mjs` with `buildEngagementSeries.test.mjs` (5 cases: warStart anchor, earliest-event fallback, per-faction grouping, zero-player filtering, and a 200k-element no-RangeError guard).

File: src/app/stats/page.jsx (hunk 5 above)
Comment (elfensky) [reply src/app/stats/page.jsx]: Fixed in 126ffc6 — added `RatioTrendChartLoader.jsx` (`next/dynamic`, `ssr: false`); `stats/page.jsx` imports the loader now, matching the /archives chart loaders so Recharts stays out of the initial bundle.

File: src/features/archives/PlayerEngagementChart.jsx (hunk 6 above)
Comment (elfensky) [reply src/features/archives/PlayerEngagementChart.jsx]: Fixed in 126ffc6 — anchor fallback now uses `withPlayers.reduce((m, e) => Math.min(m, e.start_time), Infinity)` instead of the spread. Covered by the 200k-element test.

File: src/features/stats/computeTelemetryStats.mjs (hunk 7 above)
Comment (elfensky) [reply src/features/stats/computeTelemetryStats.mjs:17]: Not changing: `seasonsWithTelemetry` counts seasons with any telemetry (`shots>0 || kills>0`), which is what the copy "seasons the bot has polled" describes. The per-chart filters (kills-only for Friendly Fire, shots-only for Accuracy) are intentional subsets. Happy to revisit if the count should track one specific chart.

File: .github/workflows/metrics.yml (hunk 2 above)
Comment (elfensky) [reply .github/workflows/metrics.yml]: Addressed in 126ffc6 — moot now that the action is back on `@v3.34`, so there is no v4 migration to validate. See the sibling thread on this line.
